### PR TITLE
Task3

### PR DESCRIPTION
**Pull request by RomanAndriienko**

### CODE REVIEW

File: Task3/pom.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            <artifactId>logback-core</artifactId>
+            <version>1.2.3</version>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>1.7.5</version>
+        </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
```
Comment (nprokhorov) [inline Task3/pom.xml]: If you are not using it, why did you add this?

File: Task3/src/main/java/com/soft/homework/model/Triangle.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class Triangle extends Shape {
+    private static final Logger logger = LoggerFactory.getLogger(Triangle.class);
+
+    private String name;
+    private double sideA;
+    private double sideB;
+    private double sideC;
```
Comment (nprokhorov) [inline Task3/src/main/java/com/soft/homework/model/Triangle.java]: Why are you saving triangle sides? 
Are they really needed?

File: Task3/src/main/java/com/soft/homework/model/Triangle.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    public Triangle(String name, double sideA, double sideB, double sideC) {
+        this.name = name;
+        this.sideA = sideA;
+        this.sideB = sideB;
+        this.sideC = sideC;
+        addSides(sideA, sideB, sideC);
+    }
+
+    private void addSides(double sideA, double sideB, double sideC) {
+        List<Double> sides = new ArrayList<>();
```
Comment (nprokhorov) [inline Task3/src/main/java/com/soft/homework/model/Triangle.java]: Does your triangle really need this list? 
Maybe three sides are enough?

File: Task3/src/main/java/com/soft/homework/model/Triangle.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
+package com.soft.homework.model;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Triangle extends Shape {
```
Comment (nprokhorov) [inline Task3/src/main/java/com/soft/homework/model/Triangle.java]: You could use `lombok` here

File: Task3/src/main/java/com/soft/homework/output/ConsoleOutput.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class ConsoleOutput implements Output {
+    private static final Logger logger = LoggerFactory.getLogger(ConsoleOutput.class);
+
+    @Override
+    public void printShape(List<ShapeData> shapes) {
```
Comment (nprokhorov) [inline Task3/src/main/java/com/soft/homework/output/ConsoleOutput.java]: Seems that now your `ConsoleOutput` knows about `shapes` ...
Maybe it would better to output some String instead of shapes

File: Task3/src/main/java/com/soft/homework/service/SortingByArea.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import com.soft.homework.data.ShapeData;
+import com.soft.homework.data.TriangleData;
+import com.soft.homework.model.Shape;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SortingByArea implements Sorting {
+    @Override
+    public List<ShapeData> sort(List<Shape> shapes) {
```
Comment (nprokhorov) [inline Task3/src/main/java/com/soft/homework/service/SortingByArea.java]: What prevents you from returning  List<Shape> shapes?

File: Task3/src/main/java/com/soft/homework/validator/creationValidator/TriangleCreationValidator.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+    @Override
+    public boolean canCreate(Shape shape) {
+        boolean can = false;
+
+        double sideA = shape.getSides().get(0);
+        double sideB = shape.getSides().get(1);
+        double sideC = shape.getSides().get(2);
+
+        if (sideA == sideB && sideC == sideA && sideC == sideB) {
```
Comment (nprokhorov) [inline Task3/src/main/java/com/soft/homework/validator/creationValidator/TriangleCreationValidator.java]: can this be simplified?

File: Task3/src/main/java/com/soft/homework/validator/inputValidator/TriangleInputValidator.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+import com.soft.homework.model.Shape;
+import com.soft.homework.model.Triangle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TriangleInputValidator implements InputValidator {
+    private static final Logger logger = LoggerFactory.getLogger(TriangleInputValidator.class);
+
+    @Override
+    public boolean validateArgs(String[] args) {
```
Comment (nprokhorov) [inline Task3/src/main/java/com/soft/homework/validator/inputValidator/TriangleInputValidator.java]: can you just pass three parameters instead of an array?

File: Task3/src/main/java/com/soft/homework/output/ConsoleOutput.java (hunk 5 above)
Comment (RomanAndriienko) [reply Task3/src/main/java/com/soft/homework/output/ConsoleOutput.java]: I understand, but I used List<ShapeData> because, if i need i can output to console list with different models iniside (triangles, squares, octagons, etc.)

File: Task3/src/main/java/com/soft/homework/service/SortingByArea.java (hunk 6 above)
Comment (RomanAndriienko) [reply Task3/src/main/java/com/soft/homework/service/SortingByArea.java]: The problem is that my abstract class Shape has an abstract method getArea(Shape shape) , and I can't sort my List<Shape> because I need to pass in parametrs shape.  And this method now can sort any models extends by shape....

File: Task3/src/main/java/com/soft/homework/validator/inputValidator/TriangleInputValidator.java (hunk 8 above)
Comment (RomanAndriienko) [reply Task3/src/main/java/com/soft/homework/validator/inputValidator/TriangleInputValidator.java]: I can do it. But I used an array in the interface, so that my validator could validate any models in the future. If it is a triangle in the array there will be a name and 3 sides, if it is a square name and 4 sides, etc.
